### PR TITLE
Add monitoring support for Postgres

### LINKED ISSUE #5: add support to health check of the database and retry mechanisms to the DB
- currently, we are not checking the health check of the DB we should check it everytime if health check is down we should retry the connection

### PR DESCRIPTION
**Pull request by Durga-chikkala**

### CODE REVIEW

**Review by DattaSaiTejaTadi — APPROVED** [review]

File: database/intializeDB.go [hunk 1]
```diff
@@ -3,49 +3,146 @@ package database
...
+	if err != nil {
+		monitoring = false
+	}
+
+	retry, err = strconv.Atoi(golf.Config.Get(prefix + "DB_RETRY_COUNT"))
+	if err != nil {
+		retry = 5
+	}
+
+	retryTime, err = strconv.Atoi(golf.Config.Get(prefix + "DB_RETRY_DURATION"))
```
Comment (babajikomali) [inline database/intializeDB.go:71]: What is DB retry duration?

File: database/intializeDB.go [hunk 2]
```diff
@@ -3,49 +3,146 @@ package database
...
+	)
+
+monitoringLoop:
+	for {
+		select {
+		case <-ticker.C:
+			if err = golf.Conn.Ping(); err != nil {
+				if retryCounter < retry {
+					for i := 0; i < retry; i++ {
+						db, err = establishDBConnection(golf.Logger, c)
```
Comment (babajikomali) [inline database/intializeDB.go:126]: What if clocks ticks between the retries, Should we increase clocking ticking duration a bit?

File: database/intializeDB.go [hunk 3]
```diff
@@ -3,49 +3,146 @@ package database
...
+						if err == nil {
+							golf.Conn = db
+							retryCounter = 0
+
+							break
+						}
+
+						retryCounter++
+						time.Sleep(time.Second * time.Duration(retryTime))
```
Comment (babajikomali) [inline database/intializeDB.go:135]: time.Sleep(time.Second * time.Duration(retryTime)) what is this doing?

File: goLF/model/golf.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package model
+
+import (
+	"database/sql"
+	"github.com/LetsFocus/goLF/configs"
```
Comment (babajikomali) [inline goLF/model/golf.go:5]: Add databases struct else it will cause errors when we raise PR for Elastic

File: database/intializeDB.go (hunk 1 above)
Comment (Durga-chikkala) [reply database/intializeDB.go:71]: It will do retry right basically this is gap between the retries

File: database/intializeDB.go (hunk 2 above)
Comment (Durga-chikkala) [reply database/intializeDB.go:126]: no need for more clarity excute the below code 

`


func  main() {

	t := time.NewTicker(time.Second * 1)

	for {
		select {
		case <-t.C:
			fmt.Println("hii")
		}

	}

	for {
		select {
		case <-t.C:
			fmt.Println("hii")
			time.Sleep(time.Second * 5)
		}
	}
}`

execute the for loops separately to understand better

**Review by babajikomali — APPROVED** [review]

**Pull request merged**